### PR TITLE
GTEST/UCP: add MT tests for ucp_put/ucp_get

### PR DESCRIPTION
**Pull request by xinzhao3**

### CODE REVIEW

**Comment by jladd-mlnx** [comment]
Looks good.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2945/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/455/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2402/ for details (Mellanox internal link).

File: test/gtest/ucp/test_ucp_tag_mt.cc [hunk 1]
```diff
@@ -82,3 +82,179 @@ UCS_TEST_P(test_ucp_tag_mt, send_recv) {
 }
 
 UCP_INSTANTIATE_TEST_CASE(test_ucp_tag_mt)
+
+
+class test_ucp_rma_mt : public test_ucp_tag_mt {
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_tag_mt.cc]: this code should be added to new file "test_ucp_rma_mt.cc" - it does not belong to "test_ucp_**tag**_mt.cc"

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/460/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2949/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2412/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/463/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/464/ for details.

**Comment by xinzhao3** [comment]
@yosefe no. I can add AMO tests asap.

**Comment by xinzhao3** [comment]
@yosefe are there any other types of gtest you want to add? I can do them all together.

**Comment by yosefe** [comment]
@xinzhao3 right, so the goal here is to make sure all UCP API functions are tested in MT mode. Please check these:
[ ] atomics
[ ] NBI rma's
[ ] request test
[ ] request cancel
[ ] request release
[ ] flush
[ ] fence

**Pull request merged**